utilities/src/bytes.rs

```diff
@@ -478,6 +478,40 @@ pub fn bytes_from_bits_le(bits: &[bool]) -> Vec<u8> {
     bytes
 }
 
+/// A wrapper around a `Write` instance that limits the number of bytes that can be written.
+pub struct LimitedWriter<W: Write> {
+    writer: W,
+    limit: usize,
+    remaining: usize,
+}
+
+impl<W: Write> LimitedWriter<W> {
+    pub fn new(writer: W, limit: usize) -> Self {
+        Self { writer, limit, remaining: limit }
+    }
+}
+
+impl<W: Write> Write for LimitedWriter<W> {
+    fn write(&mut self, buf: &[u8]) -> IoResult<usize> {
+        if self.remaining == 0 {
+            return Err(std::io::Error::new(std::io::ErrorKind::Other, format!("Byte limit exceeded: {}", self.limit)));
+        }
+
+        let max_write = std::cmp::min(buf.len(), self.remaining);
+        match self.writer.write(&buf[..max_write]) {
+            Ok(n) => {
+                self.remaining -= n;
+                Ok(n)
+            }
+            Err(e) => Err(e),
+        }
+    }
+
+    fn flush(&mut self) -> IoResult<()> {
+        self.writer.flush()
+    }
+}
+
```

Comment:
This limits the amount of bytes serialized, but that maybe we shouldn't be using it in the fundamental serialization code?

One one hand it stops people from generating transactions above the `MAX_TRANSACTION_SIZE` limit, but the devil in the details might be that if someone a node database gets an otherwise valid transaction above the size limit in the database, nodes stop being able to read their transaction stores. The potential unintended consequences here might be larger than we should risk.

However, this `Writer` DOES prevent the DoS vector of people sending large transactions that nodes constantly have to call `write_le` on. Maybe the better path is to use this serializer within a transaction `size` or `check` method on the `Transaction` struct that checks the size up to 128kb and if it's above, throws a `bail!` - and this method can be used in the checks within `synthesizer/src/vm/verify.rs`

That way we get both the benefit of limiting bytes without risking future dragons in core byte serialization.